storybookjs/storybook
# Dependency: Upgrade Yarn to v4

Fixes #24552 






## What I did

Upgrade our internal usage of Yarn to v4.

References:

- https://yarnpkg.com/blog/release/4.0
- https://github.com/yarnpkg/berry/releases/tag/%40yarnpkg%2Fcli%2F4.0.0
- https://github.com/yarnpkg/berry/issues/3591

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Checkout the branch
2. Run `git clean -xdf` (beware, will delete all uncommitted files)
3. Run `yarn start` which should take you through the whole tasks pipeline and see that it works
4. `git clean -xdf` again
5. `yarn task --task=compile --no-link` to see that it also compiles in `--no-link` mode
6. `cd scripts`
7. `yarn release:version --exact 0.0.0-canary-test-yarn.0`
8. `yarn release:publish --tag canary` to see that it correctly attempts to publish. It should output a long list of `YN0033: No authentication configured for request` errors because you don't have the npm token set, which is the correct output.

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_



### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-24565-sha-2d556c14`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24565-sha-2d556c14). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-24565-sha-2d556c14`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-24565-sha-2d556c14) |
| **Triggered by** | @JReinhold |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`upgrade-yarn-4`](https://github.com/storybookjs/storybook/tree/upgrade-yarn-4) |
| **Commit** | [`2d556c14`](https://github.com/storybookjs/storybook/commit/2d556c1426bab010edad083bc473f22c68917fcf) |
| **Datetime** | Tue Oct 24 11:57:48 UTC 2023 (`1698148668`) |
| **Workflow run** | [6626437775](https://github.com/storybookjs/storybook/actions/runs/6626437775) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=24565`_
</details>
